flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -649,14 +648,11 @@ private void configureCheckpointing() {
 		final ArrayList<MasterTriggerRestoreHook.Factory> hooks = new ArrayList<>();
 
 		for (StreamNode node : streamGraph.getStreamNodes()) {
-			if (node.getOperatorFactory() instanceof SimpleOperatorFactory) {
-				SimpleOperatorFactory factory = (SimpleOperatorFactory) node.getOperatorFactory();
-				if (factory.getOperator() instanceof AbstractUdfStreamOperator) {
-					Function f = ((AbstractUdfStreamOperator<?, ?>) factory.getOperator()).getUserFunction();
-
-					if (f instanceof WithMasterCheckpointHook) {
-						hooks.add(new FunctionMasterCheckpointHookFactory((WithMasterCheckpointHook<?>) f));
-					}
+			if (node.getOperatorFactory() instanceof UdfStreamOperatorFactory) {
+				Function f = ((UdfStreamOperatorFactory) node.getOperatorFactory()).getOperator().getUserFunction();
```

Comment:
Can not we add `getUserFunction()` method to the `UdfStreamOperatorFactory` interface?